geode-core/src/main/java/org/apache/geode/cache/query/internal/index/CompactRangeIndex.java

```diff
@@ -1555,6 +1555,9 @@ private void applyProjection(boolean add, ExecutionContext context)
         throws FunctionDomainException, TypeMismatchException, NameResolutionException,
         QueryInvocationTargetException, IMQException {
       Object indexKey = indexedExpr.evaluate(context);
+      if (IndexManager.NULL.equals(indexKey)) {
+        return;
+      }
       if (indexKey == null) {
```

Comment:
Should this still be present?  wouldn't this case be "iimpossible now that we added the three lines above?"

Reply:
Yes. This is left for the case when we are adding an entry (we are not querying) and the map does not contain the key (line 114 in CompiledIndexOperation). In that case, we don't add the index.